Hesbon5600/best-seller
# Customer signup and Login

**What does this PR do?**

Implement signup and login

**Description of Task to be completed?**

-Implement signup and login using raw user data

**How should this be manually tested?**
- Clone this repo using the command `git clone https://github.com/Hesbon5600/best-seller.git`
- Create a local copy of this branch by running `git fetch origin ft-customer-signup-login`
- Checkout to the branch `git checkout ft-customer-signup-login`
- Start the application with the command: `python manage.py runserver`
> **For signup**
- In postman, make a POST request to the endpoint `{{base-url}}/customers` with the following request body:
```source-json
{
	"email": "test-email@email.com",
	"username": "Test-Username",
	"password": "test235@Password",
	"shipping_region_id": 1
}
```
You should get a `201 Created` response like:
```source-json
{
    "customer": {
        "customer_id": 20,
        "email": "test-email@email.com",
        "username": "Test-Username",
        "shipping_region_id": 1,
        "last_login": null,
        "credit_card": null,
        "address_1": null,
        "address_2": null,
        "city": null,
        "region": null,
        "postal_code": null,
        "country": null,
        "day_phone": null,
        "eve_phone": null,
        "mob_phone": null
    },
    "accessToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6InRlc3QtZW1haWxAZW1haWwuY29tIiwiZXhwIjoxNTcwMTcxODkzfQ.MXbCOhZxonu7T4kGM5IKO9NCO-oaLcyIii6T23kLl8E",
    "expiresIn": "0:26:10.171893"
}
```
> **For Login**
- In postman, make a POST request to the endpoint `{{base-url}}/customers/login` with the following request body:
```source-json
{
	"email": "test-email@email.com",
	"password": "test235@Password",
}
```
You should get a `200 OK` response like:
```source-json
{
    "customer": {
        "customer_id": 20,
        "email": "test-email@email.com",
        "username": "Test-Username",
        "address_1": null,
        "address_2": null,
        "city": null,
        "region": null,
        "postal_code": null,
        "shipping_region_id": 1,
        "credit_card": null,
        "day_phone": null,
        "eve_phone": null,
        "mob_phone": null
    },
    "accessToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6InRlc3QtZW1haWxAZW1haWwuY29tIiwiZXhwIjoxNTcwMTcyMTE3fQ.Fc0DewZdbijk9zvIXAYJqkdg98w4MxNP8ZdK_UWp480",
    "expiresIn": "0:26:10.172117"
}
```
**Any background context you want to provide?**
NA

**What are the relevant Trello ticket?**

[Customer-login-signup](https://trello.com/c/Tm5S6M5u/2-customer-signup)


**Questions:**
NA
